MetaMask/metamask-extension
# [FLASK] Add Snaps privacy warning on snap install

## Explanation
Fixes: https://github.com/MetaMask/MetaMask-planning/issues/406

This PR will add a way to display popup with privacy policy information when installing a snap. Popup will be displayed only once when first snap is installed and never again once agreed.

Implementation is done according to the UX/UI described here: https://www.figma.com/file/diidnCHx9FXj5LDiA5DZZr/Snaps-features?type=design&node-id=1150-49137&t=z6vvUV0ZgOT98ERJ-0

## Screenshots/Screencaps
### Before
![Screenshot 2023-05-08 at 16 38 20](https://user-images.githubusercontent.com/13301024/236853433-8a999c31-667a-452e-a1df-7e51c2d43f39.png)

### After
![Screenshot 2023-05-09 at 12 20 57](https://github.com/MetaMask/metamask-extension/assets/13301024/763e9173-d305-454c-9b80-c51e886a4e9a)
![Screenshot 2023-05-09 at 12 21 16](https://github.com/MetaMask/metamask-extension/assets/13301024/3526b3df-8f37-44d9-9dd3-b339f39a930d)


## Manual Testing Steps
Scenario A:
1. Remove any previous installation of MetaMask Flask.
2. Install MetaMask Flask.
3. Install arbitrary snap.
4. Popup with notice should be displayed.
5. Click 'OK' button to agree.

Scenario B:
1. Install another arbitrary snap after first one is installed and popup is displayed.
2. Popup should not be displayed anymore whenever new snap is installed.
3. Restart browser and try again or reload the extension.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
